src/any_agent/serving/server.py

```diff
@@ -2,21 +2,33 @@
 
 from typing import TYPE_CHECKING
 
-from common.server import A2AServer
+from a2a.server.apps import A2AStarletteApplication
+from a2a.server.request_handlers import DefaultRequestHandler
+from a2a.server.tasks import InMemoryTaskStore
 
 from .agent_card import _get_agent_card
-from .task_manager import AnyAgentTaskManager
+from .agent_executor import AnyAgentExecutor
 
 if TYPE_CHECKING:
     from any_agent import AnyAgent
     from any_agent.config import ServingConfig
 
 
-def _get_a2a_server(agent: AnyAgent, serving_config: ServingConfig) -> A2AServer:
-    return A2AServer(
-        host=serving_config.host,
-        port=serving_config.port,
-        endpoint=serving_config.endpoint,
-        agent_card=_get_agent_card(agent, serving_config),
-        task_manager=AnyAgentTaskManager(agent),
+def _get_a2a_server(
+    agent: AnyAgent, serving_config: ServingConfig
+) -> A2AStarletteApplication:
+    agent_card = _get_agent_card(agent, serving_config)
+
+    request_handler = DefaultRequestHandler(
+        agent_executor=AnyAgentExecutor(agent),
+        task_store=InMemoryTaskStore(),
```

Comment:
💅 this could probably be an option to be set in `serving_config` like we have `model_type` and `DEFAULT_MODEL_TYPE` in agents.

Reply:
Since we're going to be updating the way we serve via uvicorn, let's take care of this in a future PR. https://github.com/mozilla-ai/any-agent/issues/318 to track